diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter4/VariableGenericTypeDeclarationRuleWarnTest.kt

```diff
@@ -0,0 +1,164 @@
+package org.cqfn.diktat.ruleset.chapter4
+
+import com.pinterest.ktlint.core.LintError
+import generated.WarningNames.GENERIC_VARIABLE_WRONG_DECLARATION
+import org.cqfn.diktat.ruleset.constants.Warnings
+import org.cqfn.diktat.ruleset.rules.DIKTAT_RULE_SET_ID
+import org.cqfn.diktat.ruleset.rules.VariableGenericTypeDeclarationRule
+import org.cqfn.diktat.util.LintTestBase
+import org.junit.jupiter.api.Tag
+import org.junit.jupiter.api.Test
+
+class VariableGenericTypeDeclarationRuleWarnTest : LintTestBase(::VariableGenericTypeDeclarationRule) {
```

Comment:
Will your rule trigger on something like
```kotlin
val x = foo.bar<Bar>().baz()
```
?
I mean, presence of generic parameters on the right hand side doesn't necessarily mean that property is of generic type too.